src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,83 @@
+package storage;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Represents a storage utility that can save and load serializable objects to and from a file.
+ *
+ * @param <T> The type of the object to be stored in the file.
+ */
+public class Storage<T extends Serializable> {
+    private final String relativeFilePath;
+    private final Class<T> type;
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param relativeFilePath The relative file path from the project root directory.
+     * @param type             The type of the object to be stored in the file.
+     */
+    public Storage(String relativeFilePath, Class<T> type) {
+        this.relativeFilePath = relativeFilePath;
+        this.type = type;
+    }
+
+    /**
+     * Saves an object to the file.
+     *
+     * @param item The object to be saved.
+     * @throws StorageException If there is an error saving the object to the file.
+     */
+    public void save(T item) throws StorageException {
+        this.ensureFileExists();
+        try (FileOutputStream fileOut = new FileOutputStream(this.relativeFilePath); ObjectOutputStream objectOut = new ObjectOutputStream(fileOut)) {
```

Comment:
This line seems a little long, maybe consider splitting it up?
```suggestion
        try (
            FileOutputStream fileOut = new FileOutputStream(this.relativeFilePath);
            ObjectOutputStream objectOut = new ObjectOutputStream(fileOut)
        ) {
```